xdsl/irdl/constraints.py

```diff
@@ -244,6 +264,41 @@ def can_infer(self, var_constraint_names: Set[str]) -> bool:
     def get_bases(self) -> set[type[Attribute]] | None:
         return self.constraint.get_bases()
 
+    def mapping_type_vars(
+        self, type_var_mapping: dict[TypeVar, AttrConstraint]
+    ) -> VarConstraint[AttributeCovT]:
+        return VarConstraint(
+            self.name, self.constraint.mapping_type_vars(type_var_mapping)
+        )
+
+
+@dataclass(frozen=True)
+class TypeVarConstraint(AttrConstraint):
+    """
+    Stores the TypeVar instance used to define a generic ParametrizedAttribute.
+    """
+
+    type_var: TypeVar
+    """The instance of the TypeVar used in the definition."""
+
+    constraint: AttrConstraint
+    """The base constraint of the TypeVar."""
```

Comment:
I think this should have a better name/docstring. Is this meant to represent the constraint for the bound of the typevar? Otherwise it feel strange that it is overwritten by `mapping_type_vars`.